gpytorch/distributions/multivariate_normal.py

```diff
@@ -228,6 +228,10 @@ def kl_mvn_mvn(p_dist, q_dist):
     root_p_covar = p_covar.root_decomposition()
 
     mean_diffs = q_mean - p_mean
+    if isinstance(root_p_covar, LazyTensor):
```

Comment:
Some of these changes I think are incompatible or unnecessary with the way I changed `root_decomposition` to work in https://github.com/cornellius-gp/gpytorch/pull/353. 

Maybe it would make sense to rebase this branch on top of the `lazy_root_decomp` branch so that instead of `DiagLazyTensor.root_decomposition` returning a `DiagLazyTensor` it just uses the new `root_decomposition` from that PR which accomplishes the same goal?

Reply:
@jacobrgardner is there anything blocking #353 from going in?

Reply:
@Balandat not in my opinion, no

Reply:
Do you want @gpleiss's review, or would you be happy with mine?  :)